libcaf_net/src/net/web_socket/framing.cpp

```diff
@@ -201,6 +202,10 @@ ptrdiff_t framing::handle(uint8_t opcode, byte_span payload,
     case detail::rfc6455::text_frame: {
       std::string_view text{reinterpret_cast<const char*>(payload.data()),
                             payload.size()};
+      if (!detail::rfc3629::valid(text)) {
+        abort_and_shutdown(sec::runtime_error, "invalid UTF-8 sequence");
```

Comment:
```suggestion
        abort_and_shutdown(sec::runtime_error, "invalid UTF-8 sequence");
```

I think `runtime_error` is too generic. Maybe `invalid_encoding`? We could add this as a new enumerator to `sec`. I think that would be still general enough to be useful in other contests.

Reply:
I didn't dare add a new `sec`, but yeah, I agree that this would be best. There is `malformed_basp_message`, maybe generalize this to `malformed_message` and use it everywhere, protocol agnostic?

Reply:
I like that idea. 👍

Go ahead and rename `malformed_basp_message` to `malformed_message`. It's definitely been overspecific. Also rename the enumerator in `io::basp::connection_state` to keep it consistent.